### PR TITLE
Add AnalyzedPackageVersion

### LINKED ISSUE #1298: Add AnalyzedPackageVersion
Add AnalyzedPackageVersion

### PR DESCRIPTION
**Pull request by Jericho**
This pull request was created by a tool: Cake.AddinDiscoverer version 3.57.0

Resolves #1298

### CODE REVIEW

File: extensions/Cake.AssemblyInfoReflector.yml [hunk 1]
```diff
@@ -8,4 +8,5 @@ Author: wexman
 Description: Cake addin for retrieving assembly info from an existing assembly.
 Categories:
 - assembly info
-TargetCakeVersion: 0.26.1
+TargetCakeVersion: 0.18.0
```
Comment (pascalberger) [inline extensions/Cake.AssemblyInfoReflector.yml:11]: Why the change here?

File: extensions/Cake.AutoRest.yml [hunk 2]
```diff
@@ -9,4 +9,5 @@ Description: Cake addin for generating client code from compatible API definitio
 Categories:
 - azure
 - swagger
-TargetCakeVersion: 0.18.0
+TargetCakeVersion: 0.13.0
```
Comment (pascalberger) [inline extensions/Cake.AutoRest.yml:12]: Why the change here?

File: extensions/Cake.Board.AzureBoards.yml [hunk 3]
```diff
@@ -3,7 +3,7 @@ Name: Cake.Board.AzureBoards
 NuGet: Cake.Board.AzureBoards
 Assemblies:
 - "/**/Cake.Board.AzureBoards.dll"
-Repository: https://github.com/cake-contrib/Cake.Board/
+Repository: https://github.com/nicolabiancolini/Cake.Board/
```
Comment (pascalberger) [inline extensions/Cake.Board.AzureBoards.yml:6]: Why the change here?

File: extensions/Cake.Board.AzureBoards.yml [hunk 4]
```diff
@@ -12,4 +12,5 @@ Categories:
 - work item
 - wit
 - board
-TargetCakeVersion: 0.34.1
+TargetCakeVersion: 0.33.0
```
Comment (pascalberger) [inline extensions/Cake.Board.AzureBoards.yml:15]: Why the change here?

File: extensions/Cake.DocCreator.yml [hunk 5]
```diff
@@ -3,10 +3,11 @@ Name: Cake.DocCreator
 NuGet: Cake.DocCreator
 Assemblies:
 - "/**/Cake.DocCreator.dll"
-Repository: https://github.com/agc93/DocCreator/
+Repository: https://bitbucket.org/agc93/DocCreator/
```
Comment (pascalberger) [inline extensions/Cake.DocCreator.yml:6]: Why the change here?

File: extensions/Cake.DocCreator.yml [hunk 6]
```diff
@@ -3,10 +3,11 @@ Name: Cake.DocCreator
...
 Author: agc93
 Description: A Cake addin to generate HTML documentation from Markdown source, using DocCreator
 Categories:
 - documentation
 - markdown
-TargetCakeVersion: 0.19.4
+TargetCakeVersion: 0.10.1
```
Comment (pascalberger) [inline extensions/Cake.DocCreator.yml:12]: Why the change here?

**Review by pascalberger — CHANGES_REQUESTED** [review]
@Jericho There were some changes beside the `AnalyzedPackageVersion` property being added, which seem wrong to me. What's the reason for these?

File: extensions/Cake.AssemblyInfoReflector.yml (hunk 1 above)
Comment (Jericho) [reply extensions/Cake.AssemblyInfoReflector.yml:11]: Because previously we were analyzing the most recent version of this addin (which is 1.1.0-beta2) but the logic was changed to use the most recent "stable" version which is an older version:

image

File: extensions/Cake.AutoRest.yml (hunk 2 above)
Comment (Jericho) [reply extensions/Cake.AutoRest.yml:12]: Because we changed the logic to ensure we analyze the most recent "stable" release:

image

File: extensions/Cake.DocCreator.yml (hunk 6 above)
Comment (Jericho) [reply extensions/Cake.DocCreator.yml:12]: Because we changed the logic to ensure we analyze the most recent "stable" release:

image

File: extensions/Cake.AssemblyInfoReflector.yml (hunk 1 above)
Comment (pascalberger) [reply extensions/Cake.AssemblyInfoReflector.yml:11]: Ah, didn't know you had this also already done in this version 👍

File: extensions/Cake.Board.AzureBoards.yml (hunk 3 above)
Comment (Jericho) [reply extensions/Cake.Board.AzureBoards.yml:6]: I don't know for sure. I'm guessing that's what was in the nuspec in the older package. Give me a few minutes to research...

**Review by pascalberger — APPROVED** [review]

File: extensions/Cake.Board.AzureBoards.yml (hunk 3 above)
Comment (Jericho) [reply extensions/Cake.Board.AzureBoards.yml:6]: Ok I figured it out. It's a little complicated:
1. the nuspec of the most recent **prerelease** package contains URL which points to the repo under the cake-contrib organization. This explains the `https://github.com/cake-contrib/Cake.Board/` value which was previously in the YAML
2. the nuspec in the older "stable" package contains the URL which points to the original author's repo (presumably because at that time the repo had not been transferred to the cake-contrib organization yet)
3. AddinDiscoverer has logic to check if an addin has been transferred to the cake-contrib organization and if so, automatically update the URL. In this case however, this logic fails because the name of the repo does not match the name of the addin: the repo is "Cake.Board" and the addin is "Cake.Board.AzureBoards".

File: extensions/Cake.DocCreator.yml (hunk 5 above)
Comment (Jericho) [reply extensions/Cake.DocCreator.yml:6]: I haven't done the research but my guess is that the author moved the repo from bitbucket to github between June 2016 and April 2017. That would explain the difference between these two URLs.

File: extensions/Cake.Board.AzureBoards.yml (hunk 3 above)
Comment (Jericho) [reply extensions/Cake.Board.AzureBoards.yml:6]: So in conclusion, the different values are explained because the nuspec changed between July 2019 and September 2019 AND ALSO AddinDisco was unable to automatically fix the URL because of the discrepancy between the repo name and the addin name.

Make sense?

Comment (pascalberger) [reply extensions/Cake.Board.AzureBoards.yml:6]: Yes, makes sense
